interfaces/file_storage.rb

```diff
@@ -0,0 +1,88 @@
+require_relative '../classes/book'
+require_relative '../classes/student'
+require_relative '../classes/teacher'
+require_relative '../classes/rental'
+require 'json'
+
+module FileStorage
+  BOOKS_FILE_PATH = './filestorage/books.json'.freeze
+  PERSONS_FILE_PATH = './filestorage/persons.json'.freeze
+  RENTALS_FILE_PATH = './filestorage/rentals.json'.freeze
+
+  def save_json2files(books, persons, rentals)
+    save_books(books) unless books.empty?
+    save_persons(persons) unless persons.empty?
+    save_rentals(rentals) unless rentals.empty?
+  end
+
+  def save_books(books)
+    File.new(BOOKS_FILE_PATH, 'w') unless File.exist?(BOOKS_FILE_PATH)
+    bks = []
+    books.each do |book|
+      b = { 'id' => book.id, 'title' => book.title, 'author' => book.author }
+      bks << b
+    end
+    File.write(BOOKS_FILE_PATH, JSON.generate(bks))
+  end
+
+  def save_persons(persons)
+    File.new(PERSONS_FILE_PATH, 'w') unless File.exist?(PERSONS_FILE_PATH)
+    persons.each do |person|
+      p = { 'id' => person.id, 'age' => person.age, 'name' => person.name, 'role' => person.role }
+      if person.role == 'teacher'
+        p['specialization'] = person.specialization
+      else
+        p['parent_permission'] = person.parent_permission
+      end
+      prsns << p
```

Comment:
- [ ] You applications interactions are working fine. But when I try to exit the application, the files try to save to crash because you made a typo here. The error message prints the variable `prsns` is UNdefined.
It will be nice if you please could check it out.

See screenshot.:
![a1](https://user-images.githubusercontent.com/39523312/157345367-96e2a37d-ec6d-4a38-adb1-3ae7031edddb.png)